chore(meta): Updates roles and nomination processes (#22916)
mdn/content: files/en-us/mdn/community/users_teams/index.md

---

> title: Roles and teams on MDN

Author: Keeping the order as roles first and then teams though to match the order of content.

{{MDNSidebar}}

> The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. To manage all these different groups, the project relies heavily on [roles](#roles) and [teams](#teams) in the [MDN organization on GitHub](https://github.com/mdn). A list of the members of the organization can be [found here](https://github.com/orgs/mdn/people).

Reviewer: We have 2x "There are also" sentences here, maybe we want to fix it in this PR.

Author: Done in https://github.com/mdn/content/pull/22916/commits/d34e7983411723fb927a843269909b2e2900cb76. Thanks for pointing it out.

> The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. To manage all these different groups, the project relies heavily on [roles](#roles) and [teams](#teams) in the [MDN organization on GitHub](https://github.com/mdn). A list of the members of the organization can be [found here](https://github.com/orgs/mdn/people).

Reviewer: Maybe like this?

```suggestion
The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. Changes to the site are performed by employees, contractors, and a network of partners dedicated to the health, growth, and maintenance of the project. Some individuals have committed a portion of their time to assist with running MDN Web Docs. The project relies heavily on [roles](#roles) and [teams](#teams) in the [MDN organization on GitHub](https://github.com/mdn) to manage and incorporate changes from these different groups. A list of the organization's members can be [found here](https://github.com/orgs/mdn/people).
```

Author: Oops, saw this after the previous commit. I've incorporated some of these edits. Thanks for suggestions!

- When a pull request touches files in your area of responsibility, based on the CODEOWNERS file, you will be added as a reviewer to a pull request automatically using Github's [load-balancing algorithm](https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#routing-algorithms).

- Members of a team have a higher-level repository access. Repository permissions are assigned to only those repositories where a member needs access.

The teams in the [MDN GitHub organization](https://github.com/orgs/mdn/teams) include:

- **Core**: Core MDN Web Docs team
- **mdn-community-engagement**: People responsible for community engagement across our repositories
- **mdn-product**: People responsible for the MDN Plus product
- **localization-team-leads**: People who lead our individual localization teams
- **OWD**: Contributors from the Open Web Docs non-profit organization
- **sre**: Site reliability engineers who support MDN Web Docs
- **yari-content**: The umbrella team for all MDN Web Docs content reviewers

>   - There is a sub-team for the different topic areas — accessibility, Add-ons, CSS, HTML, HTTP, Javascript, SVG, Web API, and Web assembly. For example, there's yari-content-css and yari-content-svg.

Reviewer: ```suggestion
  - There is a sub-team for the different topic areas — accessibility, Add-ons, CSS, HTML, HTTP, Javascript, SVG, Web API, and Web assembly. For example, there's `yari-content-css` and `yari-content-svg`.
```

We could also do `@yari-content-css` variant as I guess that's how they're tagged on GitHub

Author: That's a good suggestion. I added the '@' to all other team names and removed the bold formatting https://github.com/mdn/content/pull/22916/commits/fd586853a57a99b8d8afa2484cda7799f57f1c6d

>   - There are also sub-teams for different languages — Brazilian Portuguese, Chinese, French, Japanese, Korean, Russian, and Spanish. For example, there's yari-content-fr and yari-content-ko.

Reviewer: Same comment here

{{MDNSidebar}}

The success and growth of the MDN Web Docs project is, in large part, due to our community of contributors. There are also employees, contractors, and a network of partners who are dedicated to the health, growth, and maintenance of the project as a whole. There are also some individuals who have committed a portion of their time to assist with the daily tasks of running MDN Web Docs. To manage all these different groups, the project relies heavily on [roles](#roles) and [teams](#teams) within our [GitHub organization](https://github.com/mdn). A list of the members of the organization can be [found here](https://github.com/orgs/mdn/people).

Community contributions help this open source project immensely.

> In return, contributors can use their work on this project to demonstrate their writing and technical skills as well as their ability to collaborate in team settings and engage with people from diverse backgrounds. This section describes the different roles you can assume while volunteering on the MDN Web Docs project.

Reviewer: Simplify maybe?

```suggestion
Contributors can use their work on MDN Web Docs to show writing, collaboration, technical skills, and the ability to work with people from diverse backgrounds. This document describes the roles you can take while volunteering on the MDN Web Docs project.
```

Author: keeping "this section" because this document now contains a roles section and a teams section

The contributor roadmap is a step-by-step journey from one role to the next as you take on more responsibilities in the project.
With the advancement in the contributor roadmap, you could serve more than one role at the same time.

> In some cases, such as with the [Invited experts](#invited_expert) role, it's possible to skip the stepwise progression and assume a role directly if you've demonstrated expertise in a particular area.

Reviewer: Maybe simplify

```suggestion
Roles such as [Invited experts](#invited_expert) can be directly obtained if you've demonstrated expertise in a particular area.
```

## Roles

Irrespective of the role you take on in this project, you are always a [contributor](#contributor).
A **contributor** is the base role and all other roles are built on top of it.
You must satisfy the requirements of a contributor while working on this project in any capacity.

### Contributor

> Contributors, or _community participants_, engage with the project and its community by contributing their time, skills, opinions, and ideas.

Reviewer: I've already read through most of this already, the only improvement I can think of is to simplify some of the language so it's not too wordy:

```suggestion
Contributors, or _community participants_, add to the project with their time, skills, opinions, and ideas.
```

Author: Agree!!

Contributors work on the project directly and add value to it.

> The criteria for what counts as a contribution is not limited to writing or testing code, but also includes creating and updating documentation, researching, fixing bugs, and helping other community members.

Reviewer: Simplify:

```suggestion
Aside from writing and testing code, contributions include creating and updating documentation, researching, fixing bugs, and helping other community members.
```

**Requirements:**

To be eligible to be an invited expert, you must meet one or more of the following requirements:

- Demonstrated an in-depth knowledge of a particular topic area.
- Committed to being responsible for their assigned topic area.
- Supported new and occasional contributors and helped to get pull requests ready to merge.
- Attended the community meeting which takes place once every two months.

**Privileges:**

Invited experts get added to the [invited experts team](https://github.com/orgs/mdn/teams/invited-experts-and-co-maintainers) and to the appropriate topic or project team. Invited experts can:

- Access the required repository for commits and pull request approvals and merges.
- Recommend and vote for other members to become invited experts.
- Attend weekly MDN Web Docs editorial call.

### Community manager

Community managers have a distinct role in many respects.
Community managers share many of the same responsibilities as a [maintainer](#maintainer).
In addition, community managers have the following responsibilities:

> - Addressing reports of violation of code of conduct and deciding on the appropriate action.

Reviewer: A link might be helpful

```suggestion
- Addressing reports of violation of [Mozilla's code of conduct](CODE_OF_CONDUCT.md) and deciding on the appropriate action.
```

1. Go to [`Issues`](https://github.com/mdn/mdn/issues) in the `mdn/mdn` repository.
2. Click **New issue** button on the right.
3. Under 'Nominate a maintainer', click the **Get started** button.
4. Fill in the form with details of the contributions of the person you are nominating and submit the form.

<!--
TODO: Update links in these steps
TODO: Edit the section after updating the links

- The nominator will open a pull request using the appropriate [template](https://github.com/mdn/mdn-community/roles/) against the [MDN Web Docs community repository](https://github.com/mdn/mdn-community/).
- The nominee will add a comment to the pull request agreeing to all responsibilities of becoming a Maintainer.
- At least three current Maintainers must approve the pull request.
- Once the pull request is approved, the new Maintainer is added to the [Maintainers team](https://github.com/orgs/mdn/teams/maintainers) and any other relevant topic or project teams.
-->

### Nominating a spotlight contributor

See who can be a [spotlight contributor](#spotlight_contributor).
To nominate someone as a spotlight contributor, open an issue on GitHub:

> 1. Go to [`Issues`](https://github.com/mdn/mdn/issues) in the `mdn/mdn` repository.

Reviewer: Same comment: https://github.com/mdn/mdn/issues/new/choose

2. Click **New issue** button on the right.
3. Under 'Nominate a spotlight contributor', click the **Get started** button.
4. Fill in the form with details of the contributions of the person you are nominating and submit the form.

The MDN team will get in touch with the nominated person to get their information to be published on the [website](/en-US/) under "Contributor Spotlight".

### Nominating an invited expert

See who can be an [invited expert](#invited_expert).

To nominate someone as an invited expert, open an issue on GitHub:

> 1. Go to [`Issues`](https://github.com/mdn/mdn/issues) in the `mdn/mdn` repository.

Reviewer: Same 😄  https://github.com/mdn/mdn/issues/new/choose

2. Click **New issue** button on the right.
3. Under 'Nominate an invited expert', click the **Get started** button.
4. Fill in the form with details of the contributions of the person you are nominating and submit the form.

<!--
TODO: Revisit this PR process

- The nominator will open a pull request using the appropriate [template](https://github.com/mdn/mdn-community/roles/) against the [MDN Web Docs community repository](https://github.com/mdn/mdn-community/).
- The nominee will add a comment to the pull request agreeing to all responsibilities of becoming an Invited Expert.
- To be accepted, the pull request needs three approvals. This can be any combination of Invited Experts, or Maintainers.
- Once the pull request is approved, the new Invited Expert is added to the [invited-experts team](https://github.com/orgs/mdn/teams/invited-experts) and the appropriate topic team.
-->

> ### Stepping down voluntarily

Reviewer: We're talking about voluntarily stepping down in the title, but the section is about inactivity. I would rename the section to something along the lines of commitment / what activity is.

For the section content, I thought the following, but depending on the title, you can throw this away:

```
Participants must be active to set an example and show commitment to the project.
Inactivity harms the project as it may lead to unexpected delays, contributor attrition, and a loss of trust in the project.

Inactivity is defined as:

- Not contributing to the project for at least six months.
- Not responding to communication for at least three months.

Inactivity results in a community role being revoked or downgraded and the member is moved to [emeritus status](#gaining-emeritus-status).
```

> 

Reviewer: I believe we still need a "Stepping down" section where we state that it's okay to step away from roles due to any reason. We want participants to be able to let us know if they are overwhelmed, need a break, need to downgrade to a less-demanding role, or need more help, and which channels are best to talk to us etc.

Author: Valid feedback here 👍.

- I've merged the sections on "Stepping down voluntarily" and "Gaining emeritus status" under "Stepping down or applying for emeritus status"
- I've renamed "Getting demoted involuntarily" to "Demoting or removing inactive contributors" and reorganized the content in this section

Let me know what you think

Reviewer: Looks good 👍🏻

**Requirements:**

In addition to the responsibilities of other contributor roles, owners have the following responsibilities:

- Following and enforcing MDN team norms, including the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/) and [Mozilla Policies](https://www.mozilla.org/en-US/about/governance/policies/).
- Following the MDN organization policies and leading by example.
- Suggesting, documenting, and implementing new policies through the pull request process.
- Following and contributing to issues and discussions across the MDN organization.
- Ensuring that an issue or pull request gets feedback from one or more members within one week.
- [Archiving](https://help.github.com/articles/about-archiving-repositories/) or deleting unmaintained repositories.
- Discussing GitHub features, selecting the ones to use, and documenting decisions.

**Privileges:**

Owners can:

- Add and remove organization owners and members as needed.
- Add and remove collaborators to specific repositories as needed.
- Add repositories (as fresh projects or transfers) as needed.

### Summary of the roles

> ||[Contributor](#contributor)|[Organization member](#organization_member)|[Maintainer](#maintainer)|[Owner](#owner)|

Reviewer: What do you think about flipping the orientation to:

| Role                    | Requirements                                                                                                                                                                                                                                                                | Privileges                                                                                               |
| ----------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- |
| **Contributor**         | Follow the code of conduct and contribution guidelines                                                                                                                                                                                                                      | - Invitations to contributor events<br>- Eligibility to become an organization member                    |
| **Organization member** | - Enable 2FA for GitHub account<br>- Enable signed commits<br><br>One or more of:<br>- Resolve two or more issues<br>- Contribute for at least two months<br>- Active contribution in a project area                                                                        | Access rights at organization level                                                                      |
| **Maintainer**          | One or more of:<br>- Invited expert for at least six months<br>- Knowledge across multiple project areas<br>- Act towards overall health of the project<br>- Mentor other contributors<br>- Spend at least 16 hours per month on the project<br>- Attend community meetings | Approve and merge pull requests                                                                          |
| **Owner**               | Limited to Mozilla staff                                                                                                                                                                                                                                                    | - Manage access of different roles to various repositories<br>- Add or archive repositories and projects |

Author: Makes better sense this way, thanks! Fixed in https://github.com/mdn/content/pull/22916/commits/4c82f8a3a02a5c1ed6747140fea98a83db67984d

>   - **Yari content SVG**: Responsible for reviewing docs under `/en-US/docs/Web/SVG`

Reviewer: Do we need to list all of these? Is it useful for this doc?

Author: On one hand, we could remove the list and point to the mdn/teams link on GitHub. That will reduce the burden of updating this list here should any change happen to the teams (addition, removal).
On the other hand, explicitly listing the teams here might increase the likelihood of interested contributors to spot exactly the team they could help with for reviewing content (just an assumption).

Author: Valid feedback here 👍.

- I've merged the sections on "Stepping down voluntarily" and "Gaining emeritus status" under "Stepping down or applying for emeritus status"
- I've renamed "Getting demoted involuntarily" to "Demoting or removing inactive contributors" and reorganized the content in this section

Let me know what you think

Author: Thanks for the idea! Shortened the list and also updated team names as per https://github.com/orgs/mdn/teams. Fix in https://github.com/mdn/content/pull/22916/commits/565842e3c057d25994bc44cd2e71618c5cf466da.

Reviewer: Looks good!
